chat-sdk-core/src/main/java/co/chatsdk/core/utils/ImageUtils.java

```diff
@@ -48,48 +54,57 @@ public static File getDiskCacheDir(Context context, String uniqueName) {
         return new File(cachePath + File.separator + uniqueName);
     }
 
-    public static File saveImageToCache (Context context, Bitmap image) {
-        File cache = getDiskCacheDir(context, ChatSDK.config().imageDirectoryName);
-        if(!cache.exists()) {
-            cache.mkdirs();
+    public static File generateImageFile(File dir, String name, String ext) {
+        if (!dir.exists()) dir.mkdirs();
+        if (name.contains(ext)) {
+            return new File(dir, name);
         }
```

Comment:
You should point the old `saveImageToCache` function to a new function so we don't break the old API. Remember that other modules also use this code. Then you can add a deprecation annotation to the old function.